src/meta/src/hummock/manager/mod.rs

```diff
@@ -2265,3 +2280,29 @@ fn gen_version_delta<'a>(
 
     version_delta
 }
+
+async fn write_exclusive_cluster_id(
+    state_store_dir: &str,
+    cluster_id: ClusterId,
+    object_store: ObjectStoreRef,
+) -> Result<()> {
+    let cluster_id_object_path = format!("{}/{}", state_store_dir, CLUSTER_ID_OBJECT_NAME);
+    let cluster_id_res = object_store.read(&cluster_id_object_path, None).await;
+
+    if let Ok(cluster_id) = cluster_id_res {
+        Err(ObjectError::internal(format!(
+            "data directory is already used by another cluster with id {:?}",
+            String::from_utf8(cluster_id.to_vec()).unwrap()
+        ))
+        .into())
+    } else {
+        // FIXME: Can't distinguish no such item from other errors.
+        object_store
```

Comment:
So before this is fixed, a temporary object store read error may result in unexpected cluster id object overwriting, which is risky.

Reply:
I tried to use `list`, but opendal requires the path to be a directory, which behaves differently from s3.

Reply:
I have put the file in another directory called `cluster_id` and uses list to check the file's existence.